src/main/java/org/elasticsearch/index/query/QueryFilterParser.java

```diff
@@ -19,17 +19,14 @@
 
 package org.elasticsearch.index.query;
 
-import org.apache.lucene.search.Filter;
+import org.apache.lucene.search.ConstantScoreQuery;
 import org.apache.lucene.search.Query;
-import org.apache.lucene.search.QueryWrapperFilter;
 import org.elasticsearch.common.inject.Inject;
 
 import java.io.IOException;
 
-/**
- *
- */
-public class QueryFilterParser implements FilterParser {
+@Deprecated
+public class QueryFilterParser implements QueryParser {
```

Comment:
no worries about that, we will figure it out in the query-parse-refactoring branch ;)